### PR TITLE
TSK-1710: Fixed a few code smells

### PR DESCRIPTION
**Pull request by LiaLissmann**
https://sonarcloud.io/dashboard?branch=TSK-1710&id=LiaLissmann_taskana
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [x] SonarCloud analysis meets our standards
- [x] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/DbSchemaCreator.java [hunk 1]
```diff
@@ -163,8 +163,8 @@ private ScriptRunner getScriptRunnerInstance(Connection connection) {
 
   private boolean isSchemaPreexisting(Connection connection, String dbProductId) {
     ScriptRunner runner = getScriptRunnerInstance(connection);
-    StringWriter errorWriter = new StringWriter();
-    runner.setErrorLogWriter(new PrintWriter(errorWriter));
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/DbSchemaCreator.java:167]: please use `errorLogWriter`

File: common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/DbSchemaCreator.java [hunk 2]
```diff
@@ -163,8 +163,8 @@ private ScriptRunner getScriptRunnerInstance(Connection connection) {
```
Comment (mustaphazorgati) [inline common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/DbSchemaCreator.java:163]: remove that `toString()` in line 81

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java [hunk 3]
```diff
@@ -33,7 +33,6 @@
   private final List<String> orderBy;
   private final List<String> orderColumns;
 
-  private TaskHistoryQueryColumnName columnName;
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java:36]: reset

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java [hunk 4]
```diff
@@ -658,9 +657,8 @@ public TaskHistoryQuery orderByCustomAttribute(
   public List<String> listValues(
       TaskHistoryQueryColumnName dbColumnName, SortDirection sortDirection) {
     List<String> result = new ArrayList<>();
-    this.columnName = dbColumnName;
     this.orderBy.clear();
-    this.addOrderCriteria(columnName.toString(), sortDirection);
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java:663]: reset

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java [hunk 5]
```diff
@@ -33,7 +33,6 @@
   private final List<String> orderBy;
   private final List<String> orderColumns;
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/TaskHistoryQueryImpl.java:35]: annotate with `@SupressWarnings("unused");`

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java [hunk 6]
```diff
@@ -31,7 +31,6 @@
 
   private TaskanaHistoryEngineImpl taskanaHistoryEngine;
 
-  private WorkbasketHistoryQueryColumnName columnName;
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java:34]: reset

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java [hunk 7]
```diff
@@ -499,10 +498,9 @@ public WorkbasketHistoryQuery orderByOrgLevel(int num, SortDirection sortDirecti
   public List<String> listValues(
       WorkbasketHistoryQueryColumnName dbColumnName, SortDirection sortDirection) {
     List<String> result = new ArrayList<>();
-    this.columnName = dbColumnName;
     List<String> cacheOrderBy = this.orderBy;
     this.orderBy.clear();
-    this.addOrderCriteria(columnName.toString(), sortDirection);
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java:505]: reset

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java (hunk 6 above)
Comment (mustaphazorgati) [reply history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java:34]: annotate with @SupressWarnings("unused");?

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java [hunk 8]
```diff
@@ -513,7 +511,6 @@ public WorkbasketHistoryQuery orderByOrgLevel(int num, SortDirection sortDirecti
       return result;
     } finally {
       this.orderBy = cacheOrderBy;
-      this.columnName = null;
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/WorkbasketHistoryQueryImpl.java:516]: reset

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Review by benjamineckstein — APPROVED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**